src/ugenecl/src/Main.cpp

```diff
@@ -258,13 +278,34 @@ int main(int argc, char **argv) {
     // Use the first translation from the list that works.
     for (const QString &translationFile : qAsConst(translationFileList)) {
         if (translationFile == "transl_en" || translator.load(translationFile, AppContext::getWorkingDirectoryPath())) {
+            trOK = true;
             break;
         }
         failedToLoadTranslatorFiles << translationFile;
     }
-    if (!translator.isEmpty()) {
-        QCoreApplication::installTranslator(&translator);
-        GObjectTypes::initTypeTranslations();
+#ifdef Q_OS_MAC
```

Comment:
Is it correct that all we need on Mac is to check + 1 more extra dir for the translations? If yes, lets just add this dir to a common "translationSearchDirList" and use this list in the common logic above.

Reply:
The list is just a list of the filenames, and dir is current working dir
For Mac bundle the dir is NOT current working dir.
So, the code is needed to change for all files loading (in loop)
I think we don't need to change the overall code for transl loading, but just add cpesial case for mac bundle.

Reply:
This is not 100% accurate.

The list of files is a list of paths: so you can use absolute path here.

All you need is to add an extra absolute path to that list and do not call 'load'  from your code.

My suggestion:
1) Create a method in U2Private: getExtraTranslationSearchPath that will retrurn non-empty values for Mac
2) Use both in ugenecl & ugeneui to add an extra file path to the translations file list.

This way your patch to the Main.cpp will be 1 extra line of code.

Reply:
done
committed
update rebased to master

Reply:
done